var/spack/repos/builtin/packages/neurodamus-model/package.py

```diff
@@ -45,84 +44,107 @@ class NeurodamusModel(Package):
     phases = ['merge_hoc_mod', 'build', 'install']
 
     # These vars can be overriden by subclasses to specify additional sources
-    # This is required since some models have several sources, e.g.: thalamus
-    # By default they use common (which should come from submodule)
+    # This is required since some models may have a different source structure
+    # By default common mods have symlink, but hocs are copied separately
     _hoc_srcs = ('common/hoc', 'hoc')
-    _mod_srcs = ('common/mod', 'mod')
+    _mod_srcs = ('mod',)
 
-    @staticmethod
-    def copy_all(src, dst, copyfunc=shutil.copy):
-        isdir = os.path.isdir
-        for name in os.listdir(src):
-            pth = join_path(src, name)
-            isdir(pth) or copyfunc(pth, dst)
+    # The name of the mechanism, which cen be overriden
+    mech_name = ""
 
     def merge_hoc_mod(self, spec, prefix):
+        core_prefix = spec['neurodamus-core'].prefix
         # First Initialize with core hoc / mods
-        copy_tree(spec['neurodamus-core'].prefix.hoc, '_merged_hoc')
-        copy_tree(spec['neurodamus-core'].prefix.mod, '_merged_mod')
+        copy_tree(core_prefix.hoc, '_merged_hoc')
+        copy_tree(core_prefix.mod, '_merged_mod')
 
-        if spec.satisfies('+plasticity'):
-            self.copy_all('common/mod/optimized', 'common/mod')
+        # If we shall build mods for coreneuron, only bring from core those specified
+        if spec.satisfies("+coreneuron"):
+            mkdirp('core_mechs')
+            with open(core_prefix.mod.join("coreneuron_modlist.txt")) as core_mods:
+                for aux_mod in core_mods:
+                    shutil.copy(core_prefix.mod.join(aux_mod.strip()), 'core_mechs')
 
         # Copy from the several sources
         for hoc_src in self._hoc_srcs:
-            self.copy_all(hoc_src, '_merged_hoc')
+            copy_all(hoc_src, '_merged_hoc')
         for mod_src in self._mod_srcs:
-            self.copy_all(mod_src, '_merged_mod')
+            copy_all(mod_src, '_merged_mod')
+            if spec.satisfies("+coreneuron"):
+                copy_all(mod_src, 'core_mechs')
```

Comment:
Isn't this now unnecessary with symlinks?

Reply:
Neurodamusc-core is staying independent, no sym-links. So here we merge both.
I keep the for loop, but in the end mods only do one iteration now.